meshdb/meshdb.go

```diff
@@ -318,26 +328,93 @@ func (m *MeshDB) FindMiniHeaderByBlockNumber(blockNumber *big.Int) (*miniheader.
 	return miniHeaders[0], nil
 }
 
+// UpdateMiniHeaderRetentionLimit updates the MiniHeaderRetentionLimit. This is only used by tests in order
+// to set the retention limit to a smaller size, making the tests shorter in length
+func (m *MeshDB) UpdateMiniHeaderRetentionLimit(limit int) error {
+	m.MiniHeaderRetentionLimit = limit
+	return m.PruneMiniHeadersAboveRetentionLimit()
+}
+
+// PruneMiniHeadersAboveRetentionLimit prunes miniHeaders from the DB that are above the retention limit
+func (m *MeshDB) PruneMiniHeadersAboveRetentionLimit() error {
```

Comment:
Why don't we just pass in `retemtionLimit` to `PruneMiniHeadersAboveRetentionLimit`? It would let us remove a lot of code and have better separation of concerns.

Reply:
Decided to just take this on as technical debt for now. We should address it in the future, possibly at the same time as cleaning up some other parts of the MiniHeaders collection.